src/runtime/components/forms/SelectMenu.vue

```diff
@@ -423,21 +431,37 @@ export default defineComponent({
       return query.value === '' ? null : { [props.optionAttribute]: query.value }
     })
 
+    function clearOnClose () {
+      if (props.clearSearchOnClose) {
+        query.value = ''
+      }
+    }
+
+    function clearOnChange () {
+      if (props.clearSearchOnUpdate) {
+        query.value = ''
+      }
+    }
+
     watch(container, (value) => {
       if (value) {
         emit('open')
       } else {
+        clearOnClose()
         emit('close')
         emitFormBlur()
       }
     })
 
     function onUpdate (event: any) {
+      clearOnChange()
+
       if (query.value && searchInput.value?.$el) {
```

Comment:
The `query.value` is cleared anyway there.

Reply:
Yep, unnecessary